### PR TITLE
fix(react) : inconsistency of the error for showpasswordButton

### PR DESCRIPTION
**Pull request by Rogue-striker**
#### Description of changes
Added class to the ShowPasswordButton

#### Issue #, if available
Fixes https://github.com/aws-amplify/amplify-ui/issues/3581

#### Description of how you validated changes
I tested my changes in the examples folders

#### Checklist

- [x ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/types/passwordField.ts
Comment (ioanabrooks) [inline packages/react/src/primitives/types/passwordField.ts:65]: `TextFieldProps` extends `InputProps` which has this prop, so this line should not be needed.

File: packages/ui/src/theme/css/component/showPassword.scss
Comment (ioanabrooks) [inline packages/ui/src/theme/css/component/showPassword.scss:6]: Let's include this under `passwordField.scss` since this is not a separate primitive.

File: packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx
Comment (ioanabrooks) [inline packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx:37]: It would be awesome to add a unit tests/assertion to verify props are passed in as expected and error class is applied. There are examples testing these for other components, see [Checkbox.test.tsx](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/Checkbox/__tests__/Checkbox.test.tsx#L70).

File: packages/react/src/primitives/types/passwordField.ts
Comment (Rogue-striker) [reply packages/react/src/primitives/types/passwordField.ts:65]: okay

File: packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx
Comment (Rogue-striker) [reply packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx:37]: sure i'll try

File: packages/react/src/primitives/types/passwordField.ts
Comment (Rogue-striker) [reply packages/react/src/primitives/types/passwordField.ts:115]: then what about the other button props?

File: packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx
Comment (zchenwei) [inline packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx:9]: Can you move this to the top? We put third party imports on the top

File: packages/ui/src/theme/css/component/showPassword.scss
Comment (zchenwei) [reply packages/ui/src/theme/css/component/showPassword.scss:6]: +1, please do not create a new css file

File: packages/react/src/primitives/types/passwordField.ts
Comment (ioanabrooks) [reply packages/react/src/primitives/types/passwordField.ts:115]: `interface ShowPasswordButtonProps extends ButtonProps, Pick<InputProps, 'hasError'>` - can extend more than one ;)

File: packages/ui/src/theme/css/component/passwordField.scss
Comment (reesscot) [inline packages/ui/src/theme/css/component/passwordField.scss:51]: missing extra line at end of file

File: packages/ui/src/theme/css/component/passwordField.scss
Comment (zchenwei) [inline packages/ui/src/theme/css/component/passwordField.scss:46]: Could you follow the same naming convention? We are using BEM. So this should be `.amplify-passwordfield__show-password`

Comment (zchenwei) [reply packages/ui/src/theme/css/component/passwordField.scss:46]:
```suggestion
.amplify-passwordfield {
  &__show-password--error {
      color: var(--amplify-components-fieldcontrol-error-border-color);
      border-color: var(--amplify-components-fieldcontrol-error-border-color);
  }
}
```

**Review by ioanabrooks — COMMENTED** [review]
Please run unit tests locally and re-generate snapshots for failing tests.

File: packages/react/src/components/AccountSettings/ChangePassword/__tests__/__snapshots__/ChangePassword.test.tsx.snap
Comment (zchenwei) [inline packages/react/src/components/AccountSettings/ChangePassword/__tests__/__snapshots__/ChangePassword.test.tsx.snap:45]: @Rogue-striker Looking at the snapshot, I am taking back my naming suggestion coz that will be a breaking change for now. For a patch, I think we are good to use the previous name. Could you change back to use `amplify-field__show-password`? Sorry about the back and forth here :(

**Review by zchenwei — APPROVED** [review]
LGTM

**Review by zchenwei — COMMENTED** [review]
Hey @Rogue-striker, thanks for your contribution again! I tried to pull down and smoke tested it. But the style does not seem to be applied to `Button`. Could you double check locally before we get another approval and merge? 

Screenshot 2023-05-18 at 10 28 03 AM

**Comment by Rogue-striker** [comment]
it should work now. can you please check once again

File: packages/react/src/primitives/PasswordField/__tests__/PasswordField.test.tsx
Comment (reesscot) [inline packages/react/src/primitives/PasswordField/__tests__/PasswordField.test.tsx:118]:
```suggestion
  it('should show error styling when field has error', async () => {
```

File: packages/ui/src/theme/css/component/passwordField.scss
Comment (reesscot) [inline packages/ui/src/theme/css/component/passwordField.scss:1]: Do we know why this changed? It appears to be identical. Was it just formatting?

**Review by dbanksdesign — DISMISSED** [review]
Changes have been verified:

| Live | PR |
| - | - |
| CleanShot 2023-06-01 at 12 29 31@2x | CleanShot 2023-06-01 at 12 29 00@2x |

**Review by reesscot — APPROVED** [review]
LGTM, thanks for your contribution @Rogue-striker!

File: packages/react/src/primitives/types/passwordField.ts
Comment (zchenwei) [inline packages/react/src/primitives/types/passwordField.ts:67]: To resolve the conflicts, this should be 
```suggestion
  extends BaseButtonProps,
    Pick<BaseInputProps, 'hasError'> {
```

File: packages/react/src/primitives/types/passwordField.ts
Comment (zchenwei) [inline packages/react/src/primitives/types/passwordField.ts:4]:
```suggestion
import { BaseInputProps } from './input';
```

File: packages/ui/src/theme/css/component/passwordField.scss
Comment (Rogue-striker) [reply packages/ui/src/theme/css/component/passwordField.scss:1]: it formatting only i guess

**Review by hbuchel — APPROVED** [review]
Smoke tested locally and LGTM!

**Review by reesscot — APPROVED** [review]

**Review by dbanksdesign — APPROVED** [review]
LGTM!

**Comment by hbuchel** [comment]
@Rogue-striker Thank you for your contribution and for your patience with us getting this merged!

**Comment by offlineprogrammer** [comment]
@Rogue-striker Thank you for your contribution.  Can you please connect with me to claim your badge

**Pull request merged**
